service/src/main/java/org/cbioportal/service/impl/CoExpressionServiceImpl.java

```diff
@@ -99,13 +139,10 @@
                 continue;
             }
             
-            double[][] arrays = new double[valuesNumber.length][2];
-            for (int i = 0; i < valuesNumber.length; i++) {
-                arrays[i][0] = queryValuesNumber[i];
-                arrays[i][1] = valuesNumber[i];
-            }
-
-            SpearmansCorrelation spearmansCorrelation = new SpearmansCorrelation(new Array2DRowRealMatrix(arrays, false));
+            double[][] arrays = new double[2][valuesNumber.length];
+            arrays[0] = queryValuesNumber;
+            arrays[1] = valuesNumber;
+            SpearmansCorrelation spearmansCorrelation = new SpearmansCorrelation((new Array2DRowRealMatrix(arrays, false)).transpose());
```

Comment:
@jjgao @n1zea144 @lemccarthy  unless I'm mistaken, it would seem that at minimum we need to bring into memory all alterations for a single gene in order to populate the array we pass to SpearmansCorrelation here.   If that's true, does a cursor help us?  I think a cursor is for handling db result record by record.  But we cannot do that, right?